### DISCUSSION TITLE
计算器的问题(..•˘_˘•..)

### DISCUSSION THREAD

**Discussion by PureDimension** [liusy58/NJU_AdvancedProgramming_SP22#20] [Q&A] [answered]
1. 关于报错问题（我只写完一个检查模块就已经200行了…………）
如果式子有多个错误，是不是只要呈现第一个发现的错误就可以？
2.式子化简的时候有没有限定要求？比如买了3个12元和12个3元，要求简化成3x(12+12),(3+3)x12,不简化还是都行？
3.ppt上计算器那一页给的非法输入例子括号是中文括号诶，应该是打错了吧hhh
4.百度查有个叫stoi的东西可以用吗？孩子字符串转数字头疼一学期了每次都手动写他个一百行来转（）

还有代码长度不会影响评分吧，预感代码会很长hmm

**Comment by Price1999a** [comment, marked as answer]
1 是的

3 没错

4 可以用stl的函数

2 ppt说是按照相同个数进行化简

**Reply by PureDimension** [reply]
感谢(´-ω･ˋ)

**Reply by PureDimension** [reply]
话说相同个数的话，是不是默认个数排在前面而价格排在后面，比如3x12一定表示买3个12块钱的www

**Reply by binbin00** [reply]
合并只是为了产生圆括号，其余的请不必过多纠结。这块的重点在于你的计算器 能否判断表达式是否合法 以及 计算合法表达式的值。

**Reply by PureDimension** [reply]
> 



> 合并只是为了产生圆括号，其余的请不必过多纠结。这块的重点在于你的计算器 能否判断表达式是否合法 以及 计算合法表达式的值。

好哒ww
